source/agora/consensus/protocol/Nominator.d

```diff
@@ -544,6 +552,14 @@ extern(D):
                     fail_reason, data);
                 return ValidationLevel.kInvalidValue;
             }
+
```

Comment:
You could do one more check here. If the array is bigger than the total number of active validators, then it's clearly invalid. So in that case you can return `kInvalidValue`.

Reply:
> You could do one more check here. If the array is bigger than the total number of active validators, then it's clearly invalid. So in that case you can return kInvalidValue.

Ah, Right. It should return `kInvalidValue` in that case. I will fix it with the other requirements you suggested.
Thank you.